### ISSUE TITLE
Add support for JAXRS sub-resources

### ISSUE THREAD

**Issue by nhartner** [teamcarma/swagger-jaxrs-doclet#38]
I would like to see support for sub-resources/resource locators: https://jersey.java.net/documentation/latest/jaxrs-resources.html#d0e2184

swagger-core has a sample project that uses sub-resources: https://github.com/swagger-api/swagger-core/tree/master/samples/java-jaxrs-subresource

**Comment by conorroche** [comment]
@nhartner sub resources are supported. I have test cases that cover 2 levels of subresources. Is there a specific example you can give me that is not working?

**Comment by nhartner** [comment]
Ah ok, I see that it works if you return an instance of a subresource instead of the subresource class.  Jersey supports both.  For example, here are 2 different ways you can do a subresource method:

```java
@Path("/greetings")
public class GreetingsResource {

    @Path("/hello")
    public HelloWorldSubResource getHello() {
      return new HelloWorldSubResource();
    }

    @Path("/helloworld")
    public Class<HelloWorldSubResource> getHelloWorld() {
      return HelloWorldSubResource.class;
    }

    public static class HelloWorldSubResource {
      @GET
      public String getHello() {
        return "hello";
      }
    }
}
```

Both /greetings/hello and /greetings/helloworld will invoke the subresource as expected.  The doclet however only shows /greetings/hello in the API docs but not /greetings/helloworld. It would be nice if the doclet recognized the latter method.

**Comment by conorroche** [comment]
ive just committed a fix for this as part of the latest 1.0.4-SNAPSHOT

**Comment by conorroche** [comment]
@nhartner can you confirm if this has solved your problem?

**Comment by nhartner** [comment]
Awesome.  Latest snapshot looks good on my app.

**Comment by TuomasKiviaho** [comment]
I've a similar use case as above with ``@POST`` method. Only difference is that Resteasy seems to require ``@Path`` on this method and that the subresource class I use is not an inner class. This class is considered also as normal resource by the doclet but not by Resteasy. Could the extra ``@Path`` declaration on the method be the culprit here.

**Comment by conorroche** [comment]
no i have test cases that have @path on the method, do you by any chance have @path on the sub resource class?

**Comment by TuomasKiviaho** [comment]
@conorroche Nope, that's the key difference between sub resource and real resource (as I understand it). If I replace``@Path("/post")`` as ``@Path("/")`` then I'm getting ``SwaggerOperation bar is missing path.`` because the output becomes``"path" : ""``. I break Resteasy when I remove the annotation completely.

```
public abstract class Bar {

    @POST
    @Path("/post")
    public abstract void bar();

}

@Path("/foo")
@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
public abstract class Foo {

    @Path("/bar")
    @Produces(MediaType.APPLICATION_JSON)
    public abstract Bar getBar();

}
```

**Comment by conorroche** [comment]
@TuomasKiviaho i have committed a fix for this against issue #46 can you let me know if this resolves your issue, thanks.
